### PR TITLE
Optimization for Float floor and ceil functions when precision is 0

### PR DESCRIPTION
**Pull request by amatalai**
#### Some benchmarks:
Float ceil/floor 1000 times per benchmark run

**Ceil**

Before changes
```
Name                         ips        average  deviation         median         99th %
ceil/1                    369.95     2703.04 μs     ±9.48%        2615 μs        3897 μs
ceil/2 precision 0        366.37     2729.45 μs     ±8.96%        2674 μs     3743.83 μs
ceil/2 precision 1        350.72     2851.25 μs     ±4.61%        2821 μs     3356.70 μs
```

After changes
```
Name                         ips        average  deviation         median         99th %
ceil/1                  23214.94       43.08 μs    ±17.66%          42 μs          81 μs
ceil/2 precision 0      22786.86       43.88 μs    ±18.15%          42 μs          81 μs
ceil/2 precision 1        364.20     2745.76 μs     ±4.75%        2709 μs     3260.11 μs
```

**Floor**

Before changes
```
Name                         ips        average  deviation         median         99th %
floor/1                    382.26     2616.04 μs     ±4.29%        2591 μs     3050.78 μs
floor/2 precision 0        374.95     2667.04 μs     ±5.65%        2624 μs        3353 μs
floor/2 prevision 1        353.21     2831.22 μs     ±4.42%        2808 μs     3310.42 μs
```

After changes
```
Name                          ips        average  deviation         median         99th %
floor/1                  23373.19       42.78 μs    ±15.30%          42 μs          80 μs
floor/2 precision 0      22528.48       44.39 μs    ±16.96%          43 μs          80 μs
floor/2 prevision 1        359.11     2784.66 μs     ±4.00%        2762 μs     3203.20 μs
```

#### Property based tests with StreamData
https://github.com/amatalai/float_optimization/blob/master/test/float_optimization_test.exs

### CODE REVIEW

**Comment by michalmuskala** [comment]
How does using `:math.floor` and `:math.ceil` compare?

File: lib/elixir/lib/float.ex [hunk 1]
```diff
@@ -149,6 +149,11 @@ defmodule Float do
   @spec floor(float, precision_range) :: float
   def floor(number, precision \\ 0)
 
+  def floor(number, 0) when is_float(number) do
+    int = :erlang.floor(number)
```
Comment (sorentwo) [inline lib/elixir/lib/float.ex]: This seems like a perfect use of pipes, is there any reason not to use them here?

```elixir
number
|> :erlang.floor()
|> :erlang.float()
```

**Comment by amatalai** [comment]
`:math.ceil/1` and `:math.floor/1` are faster

```
Name                         ips        average  deviation         median         99th %
erlang ceil             23214.94       43.08 μs    ±17.66%          42 μs          81 μs
math ceil               39032.17       25.62 μs    ±30.36%          25 μs          48 μs

erlang floor            23373.19       42.78 μs    ±15.30%          42 μs          80 μs
math floor              46143.92       21.67 μs    ±21.94%          21 μs          39 μs
```

Also, according to benchee `:math` version uses less memory

thank you @michalmuskala

**Comment by amatalai** [comment]
My first commit doesn't make much sense now. Should I rebase them and force push?

**Comment by josevalim** [comment]
We squash when merging. :)
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**